### PR TITLE
feat(api): add api spec for exposing site metrics by source and implementation

### PR DESCRIPTION
**Pull request by iuliag**
Required for SITES-21063

Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description. Or if there's no issue created, make sure you 
  describe here the problem you're solving.
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

If the PR is changing the API specification:
- [x] make sure you add a "Not implemented yet" note the endpoint description, if the implementation is not ready 
  yet. Ideally, return a 501 status code with a message explaining the feature is not implemented yet.
- [x] make sure you add at least one example of the request and response.

If the PR is changing the API implementation or an entity exposed through the API:
- [ ] make sure you update the API specification and the examples to reflect the changes.

## Related Issues

Thanks for contributing!

### CODE REVIEW

**Comment by solaris007** [comment]
@iuliag will this API be flagged in some way, e.g. "prototype / temporary" as it is not a validated customer value use case but rather an "internal" COGS-saving tool?

**Comment by iuliag** [comment]
I've added the `experimental` tag, to mark that this API is prototype / temporary.

File: docs/openapi/schemas.yaml [hunk 1]
```diff
@@ -898,3 +898,44 @@ SiteTopPageList:
...
+    metric:
+      description: The name of the metric
+      type: string
+    value:
+      description: The value of the metric
+      type: integer
+    time:
+      description: The date and time when the metric was recorded
+      $ref: '#/DateTime'
+    granularity:
```
[the remark this answers is not in the document]
Comment (iuliag) [reply docs/openapi/schemas.yaml:925]: Yes, if you look at 2 consecutive items in the list, the diff between the timestamps should give you the granularity.

[the remark this answers is not in the document]
Comment (solaris007) [reply docs/openapi/schemas.yaml:925]: as such i'd start minimally and drop the granularity for now.

**Review by solaris007 — APPROVED** [review]
LGTM, minor comment

**Review by solaris007 — APPROVED** [review]

**Pull request merged**